script.js

```diff
@@ -1,4 +1,709 @@
+var numberOfDice = UNASSIGNED;
+// Create global variable to store the number of players indicated at the start of the game
+var numberOfPlayers = ``;
+
+// Create function to check input is a number > 0
+var checkNumber = function (data) {
+  return !isNaN(Number(data)) && Number(data) != 0;
+};
+
+// Create a function for random dice roll
+var diceRoll = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+// Create a function to roll a specific quantity of dice stated
+var rollAnyQtyDice = function (quantity) {
+  diceRollArray = [];
+  var counter = 0;
+  while (counter < quantity) {
+    diceRollArray.push(diceRoll());
+    counter += 1;
+  }
+  return diceRollArray;
+};
+
+// Sort an array in an ascending order
+var sortAsc = function (array) {
+  return array.sort(function (a, b) {
+    return a - b;
+  });
+};
+// Sort an array in a descending order
+var sortDes = function (array) {
+  return array.sort(function (a, b) {
+    return b - a;
+  });
+};
+
+// Store the unsequenced dice numbers rolled in the current round for the current player
+var storeRandomArray = function (array) {
+  var diceCounter = 0;
+  var unsortedarray = ``;
+  while (diceCounter < array.length) {
+    unsortedarray = String(unsortedarray) + String(array[diceCounter]);
+    diceCounter += 1;
+  }
+  return unsortedarray;
+};
+
+// Auto-generate highest/lowest combined number based on result mode
+var autoCombine = function (resultMode, array) {
+  if (resultMode == NORMAL) {
+    array = sortDes(array);
+    var bestNumber = ``;
+    itemCounter = 0;
+    while (itemCounter < array.length) {
+      bestNumber = String(bestNumber) + String(array[itemCounter]);
+      itemCounter += 1;
+    }
+    console.log("bestNumber", bestNumber);
+    return bestNumber;
+  }
+  if (resultMode == REVERSED) {
+    var array = sortAsc(array);
+    var bestNumber = ``;
+    itemCounter = 0;
+    while (itemCounter < array.length) {
+      bestNumber = String(bestNumber) + String(array[itemCounter]);
+      itemCounter += 1;
+    }
+    console.log("bestNumber", bestNumber);
+    return bestNumber;
+  }
+};
+
+// Find the running sum of the numbers in an array
+var addTotal = function (array) {
+  var total = ``;
+  counter = 0;
+  while (counter < array.length) {
+    total = Number(total) + Number(array[counter]);
+    counter += 1;
+  }
+  return total;
+};
+
+// Find winning player based on result mode (normal/reversed)
+// Set 'Player 1' as default winner
+var winnersGreatest = [1];
+var winnersSmallest = [1];
+var findWinner = function (array, normalOrReversed) {
+  // Generate updated list of winning players
+  if (
+    // Do not need to compare if the player is already in the winners list since the previous round
+    !winnersGreatest.includes(playerNo + 1) &&
+    // Add to list if the player's number is same as that of current winner(s)
+    array[[winnersGreatest[0]] - 1] == array[playerNo]
```

Comment:
Reason why is because having something like this `array[winnersGreatest[0]] - 1]`
would cause a couple of potential problems when revisiting and debugging.